x/tokenomics/module/abci.go

```diff
@@ -59,12 +59,19 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) (err error) {
 
 	// Update the relay mining difficulty for every service that settled pending
 	// claims based on how many estimated relays were serviced for it.
-	err = k.UpdateRelayMiningDifficulty(ctx, relaysPerServiceMap)
+	difficultyPerServiceMap, err := k.UpdateRelayMiningDifficulty(ctx, relaysPerServiceMap)
 	if err != nil {
 		logger.Error(fmt.Sprintf("could not update relay mining difficulty due to error %v", err))
 		return err
 	}
 	logger.Info(fmt.Sprintf("successfully updated the relay mining difficulty for %d services", len(relaysPerServiceMap)))
 
+	// Emit telemetry for each service's relay mining difficulty.
+	for serviceId, newDifficulty := range difficultyPerServiceMap {
+		miningDifficultyNumBits := keeper.RelayMiningTargetHashToDifficulty(newDifficulty.TargetHash)
+		telemetry.RelayMiningDifficultyGauge(miningDifficultyNumBits, serviceId)
+		telemetry.RelayEMAGauge(newDifficulty.NumRelaysEma, serviceId)
+	}
+
```

Comment:
Any reason not to defer this? I'm imagining a future change adds some code after this which introduces a new error branch. I don't think it's necessary to declare `difficultyPerServiceMap` earlier or move the for loop up, but let's at least protect ourselves ... from ourselves. :sweat_smile: